### PR TITLE
Perform `insertMany` in a single transaction

### LINKED ISSUE #2510: Perform `insertMany` in a single transaction
### What should be done?

Currently we perform insertion of multiple documents by calling `insertDocument` function multiple times. The function creates transaction for each document.

Let's try to perform all of the inserts in a single transaction. That might remove some overhead related to the transaction and potentially decrease the insert time.

Merging everything into single transaction will break compatibility, as on error - PostgreSQL will "undo" all previous inserts, while MongoDB won't. In such scenarios we should perform insertions once again, but in separate transactions.

Let's start by introducing simple insertion benchmark in separate PR. It will allow us to check if there's any performance gain after second PR with the actual change.

## Where
`pgdb` package

## DoD
- [x] Benchmark for `insertMany` created
- [x] Inserts of many documents are performed in a single transaction
- [x] If any of documents failed on insert, insertion is performed once again in separate transactions
- [x] Benchmarks results are compared.

### PR DESCRIPTION
**Pull request by raeidish**
# Description

Closes #2510 

## Readiness checklist
* [ ] I added/updated unit tests.
* [ ] I added/updated integration/compatibility tests.
* [ ] I added/updated comments and checked rendering.
* [x] I made spot refactorings.
* [ ] I updated user documentation.
* [ ] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/handlers/pg/msg_insert.go [hunk 1]
```diff
@@ -95,6 +95,33 @@ func (h *Handler) MsgInsert(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
...
+
+	dbPool.InTransaction(ctx, func(tx pgx.Tx) error {
+        var err error
+        for i := 0; i < docs.Len(); i++ {
+            doc := must.NotFail(docs.Get(i))
+            err = insertDocument(ctx, dbPool, qp, doc)
+            var we *commonerrors.WriteErrors
+            switch {
+            case err == nil:
+                inserted++
```
Comment (rumyantseva) [inline internal/handlers/pg/msg_insert.go:110]: In principle, we don't need to calculate it: as it's a transition, we insert all or nothing.

Comment (raeidish) [reply internal/handlers/pg/msg_insert.go:110]: Yes that is true, i will discard it in the single transaction run

**Comment by noisersup** [comment]
Is there anything I can help with, or is this one ready for review?

**Comment by AlekSi** [comment]
(more specifically, we are not sure due to PR title 😄)

**Comment by raeidish** [comment]
Aha! Sorry I’m a bit busy at the moment, the pr is ready for review. There was one suggestion where I would revert the changes to find and modify and update calls. This is because of the update function might return incorrect error messages

**Review by noisersup — APPROVED** [review]
Looks great! I've tested it locally and there benchmark results have improved.

**Pull request merged**